consensus/safety-rules/src/safety_rules.rs

```diff
@@ -302,16 +302,20 @@ impl SafetyRules {
             .cloned()
             .ok_or(Error::InvalidLedgerInfo)?;
 
+        // Update the waypoint to a newer value, this might still be stale than the current epoch.
```

Comment:
Can we put this into an if statement, no point in updating to the same value. I guess the generated waypoint should always be equal or newer, otherwise line 297 would fail?

```suggestion
        // Update the waypoint to a newer value, this might still be older than the current epoch.
```